flink-kubernetes-operator-autoscaler/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/AutoscalerFlinkMetrics.java

```diff
@@ -62,54 +61,49 @@ public AutoscalerFlinkMetrics(MetricGroup metricGroup) {
-                            LOG.info("Registering scaling metrics for job vertex {}", jobVertexID);
-                            var jobVertexMg =
-                                    metricGroup.addGroup(JOB_VERTEX_ID, jobVertexID.toHexString());
-
-                            evaluated.forEach(
-                                    (sm, esm) -> {
-                                        var smGroup = jobVertexMg.addGroup(sm.name());
-
-                                        smGroup.gauge(
-                                                CURRENT,
-                                                () ->
-                                                        Optional.ofNullable(
-                                                                        currentVertexMetrics.get())
-                                                                .map(m -> m.get(jobVertexID))
-                                                                .map(metrics -> metrics.get(sm))
-                                                                .map(
-                                                                        EvaluatedScalingMetric
-                                                                                ::getCurrent)
-                                                                .orElse(Double.NaN));
-
-                                        if (sm.isCalculateAverage()) {
-                                            smGroup.gauge(
-                                                    AVERAGE,
-                                                    () ->
-                                                            Optional.ofNullable(
-                                                                            currentVertexMetrics
-                                                                                    .get())
-                                                                    .map(m -> m.get(jobVertexID))
-                                                                    .map(metrics -> metrics.get(sm))
-                                                                    .map(
-                                                                            EvaluatedScalingMetric
-                                                                                    ::getAverage)
-                                                                    .orElse(Double.NaN));
-                                        }
-                                    });
-                        });
+                    metricsSupplier) {
+        if (scalingMetricsInitialized) {
+            // It is important that we only initialize these metrics once because the Flink API does
+            // not support registering counters / gauges multiple times.
+            // The metrics will be updated via the provided metricsSupplier.
+            return;
+        }
+        scalingMetricsInitialized = true;
+
+        LOG.info("Registering scaling metrics");
```

Comment:
Is this log necessary?

If yes, could we add the resourceId(similar to job information)? Otherwise we don't know which job is registering scaling metrics.

Reply:
The logger is already scoped by resource id.

Reply:
The logging was there before and I think it is helpful to print it once after resource creation.